### PR TITLE
Synchronize manual with stubs for PHP 8.4 - part 3

### PR DESCRIPTION
**Pull request by kocsismate**

### CODE REVIEW

**Review by cmb69 — COMMENTED** [review]
Thank you!

**Review by cmb69 — APPROVED** [review]

File: reference/dir/directory.xml [hunk 1]
```diff
@@ -85,6 +86,12 @@
       </row>
      </thead>
      <tbody>
+      <row>
+       <entry>8.4.0</entry>
+       <entry>
+        This class is now <literal>final</literal>.
+       </entry>
+      </row>
```
Comment (Girgias) [inline reference/dir/directory.xml:94]: Incorrect this is part of https://wiki.php.net/rfc/directory-opaque-object which is PHP 8.5.

Please use a worktree and the stubs from PHP-8.4 to do these changes.

File: reference/zip/ziparchive.xml [hunk 2]
```diff
@@ -480,6 +480,12 @@
      <type>int</type>
      <varname linkend="ziparchive.constants.er-not-allowed">ZipArchive::ER_NOT_ALLOWED</varname>
     </fieldsynopsis>
+    <fieldsynopsis>
+     <modifier>public</modifier>
+     <modifier>const</modifier>
+     <type>int</type>
+     <varname linkend="ziparchive.constants.er-truncated-zip">ZipArchive::ER_TRUNCATED_ZIP</varname>
+    </fieldsynopsis>
```
Comment (Girgias) [inline reference/zip/ziparchive.xml:488]: This needs a description below, also available as of PHP 8.4.0. (refer back to the issue tracking these changes)

File: reference/soap/soapclient.xml [hunk 3]
```diff
@@ -257,42 +257,42 @@
...
-    <varlistentry xml:id="soapclient.props.--last-response-headers">
+    <varlistentry xml:id="soapclient.props.last-response-headers">
      <term><varname>__last_response_headers</varname></term>
      <listitem>
       <para>
       </para>
      </listitem>
     </varlistentry>
-    <varlistentry xml:id="soapclient.props.--soap-fault">
+    <varlistentry xml:id="soapclient.props.soap-fault">
```
Comment (Girgias) [inline reference/soap/soapclient.xml:295]: Not sure if those soap changes are correct, I'm pretty sure you are breaking links with this rn.

**Review by Girgias — CHANGES_REQUESTED** [review]

File: reference/zip/ziparchive.xml (hunk 2 above)
Comment (kocsismate) [reply reference/zip/ziparchive.xml:488]: The description has already been added previously :)

File: reference/dir/directory.xml (hunk 1 above)
Comment (kocsismate) [reply reference/dir/directory.xml:94]: Ah, I forgot that this was voted for PHP 8.5. Problem with using an older branch when generating the manual pages is that gen_stub.php usually receives improvements in master only, therefore I usually just disregard the stub changes that are not yet relevant.

File: reference/zip/ziparchive.xml (hunk 2 above)
Comment (kocsismate) [reply reference/zip/ziparchive.xml:488]: https://www.php.net/manual/en/zip.constants.php#ziparchive.constants.er-truncated-zip for reference

Comment (Girgias) [reply reference/zip/ziparchive.xml:488]: Okay so this item is actually done, surprising.

File: reference/soap/soapclient.xml (hunk 3 above)
Comment (Girgias) [reply reference/soap/soapclient.xml:295]: Can you check if those IDs are used in a `linken` attribute or not? But in the same vein, you should remove the single hyphen from the constants below.

File: reference/dir/directory.xml (hunk 1 above)
Comment (Girgias) [reply reference/dir/directory.xml]: But you could use the stub files from PHP-8.4 with master gen_stub, no?

File: reference/soap/soapclient.xml (hunk 3 above)
Comment (kocsismate) [reply reference/soap/soapclient.xml:295]: I found no collision. In the same time, I removed the leading single hyphens.

**Review by Girgias — APPROVED** [review]

**Pull request merged**